aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,459 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Test that the current implementation of Auth as a wrapper of AWSMobileClient calls the correct
+ * AWSMobileClient methods with the correct parameters when the different Auth methods are called.
+ */
+@RunWith(RobolectricTestRunner.class)
+@SuppressWarnings("unchecked")
+public final class AuthComponentTest {
+    private AWSMobileClient mobileClient;
+    private AWSCognitoAuthPlugin authPlugin;
+    private final String userId = "myId";
+    private final String destination = "e***@email.com";
+    private final String deliveryMedium = "sms";
+    private final String attributeName = "email";
+    private final String username = "username";
+    private final String password = "password123";
+    private final String newPassword = "newPassword123";
+    private final String attributeKey = AuthUserAttributeKey.email().getKeyString();
+    private final String attributeVal = "email@email.com";
+    private final String confirmationCode = "confirm";
+    private final Map<String, String> metadata = Collections.singletonMap("aCustomKey", "aCustomVal");
+
+    @Before
+    public void setup() throws AmplifyException {
+        mobileClient = mock(AWSMobileClient.class);
+        authPlugin = new AWSCognitoAuthPlugin(mobileClient, userId);
+    }
+
+    /**
+     * Tests that the configure method of the Auth wrapper of AWSMobileClient (AMC) calls AMC.initialize() with the
+     * passed in context, a new AWSConfiguration object containing the passed in JSONObject, and a callback object
+     * which causes configure to complete successfully in the onResult case.
+     * @throws AmplifyException an exception wrapping the exception returned in onError of AMC.initialize()
+     * @throws JSONException has to be declared as part of creating a test JSON object
+     */
+    @Test
+    public void testConfigure() throws AmplifyException, JSONException {
+        UserStateDetails userStateDetails = new UserStateDetails(UserState.SIGNED_OUT, null);
+        Context context = getApplicationContext();
+        JSONObject testConfig = new JSONObject().put("TestKey", "TestVal");
+
+        doAnswer(invocation -> {
+            assertEquals(context, invocation.getArgument(0));
+            assertEquals(testConfig.toString(), invocation.getArgument(1).toString());
+
+            Callback<UserStateDetails> callback = invocation.getArgument(2);
+            callback.onResult(userStateDetails);
+            return null;
+        }).when(mobileClient).initialize(any(Context.class), any(AWSConfiguration.class), any(Callback.class));
+
+        authPlugin.configure(testConfig, context);
+        verify(mobileClient, times(1))
```

Comment:
`times(1)` is the default, I believe. So you can just say `verify(foo).gotFooed()`. The `times(int)` comes into play when the value is _not_ `1`.

Reply:
Aha, good to know